lib/graphql/schema.rb

```diff
@@ -221,6 +221,22 @@ def types
       end
     end
 
+    # Returns a list of Arguments and Fields referencing a certain type
+    # @param type_name [String]
+    # @return [Hash]
+    def get_members_of_type(type_name)
+      rebuild_artifacts unless defined?(@type_reference_map)
+      @type_reference_map.fetch(type_name, [])
+    end
+
+    # Returns a list of Union types in which a type is a member
+    # @param type [GraphQL::ObjectType]
+    # @return [Array<GraphQL::UnionType>] The field named `field_name` on `parent_type`    def union_memberships(type)
+    def union_memberships(type)
+      rebuild_artifacts unless defined?(@unions)
+      @unions.select { |union| possible_types(union).include?(type) }
```

Comment:
Can we cache this calculation? For example, could `@unions` be a 

```ruby
Hash.new { |h, k| h[k] = [] }
```

Where the keys are object types and the values are lists of union types which contain that object type? 

(A similar cache is built by `warden.rb`, but could we cache it at schema-level instead/also?)